Content.Server/GameTicking/Rules/ZombieRuleSystem.cs

```diff
@@ -263,82 +397,169 @@ private List<EntityUid> GetHealthyHumans(bool includeOffStation = true)
     ///     allowing this gamemode to be started midround. As such, it doesn't need
     ///     any information besides just running.
     /// </remarks>
-    private void InfectInitialPlayers(ZombieRuleComponent component)
+    private void InfectInitialPlayers(EntityUid uid, ZombieRuleComponent rules)
     {
-        if (component.InfectedChosen)
-            return;
-        component.InfectedChosen = true;
 
         var allPlayers = _playerManager.ServerSessions.ToList();
         var playerList = new List<IPlayerSession>();
         var prefList = new List<IPlayerSession>();
+
         foreach (var player in allPlayers)
         {
-            if (player.AttachedEntity == null || !HasComp<HumanoidAppearanceComponent>(player.AttachedEntity))
-                continue;
-            playerList.Add(player);
+            // TODO: A
+            if (player.AttachedEntity != null && HasComp<HumanoidAppearanceComponent>(player.AttachedEntity))
+            {
+                if (TryComp<ZombieComponent>(player.AttachedEntity, out var zombie))
+                {
+                    // This player is already a zombie. If they don't already have a rule, add them to this one.
+                    if (zombie.Family.Rules == EntityUid.Invalid)
+                    {
+                        zombie.Family.Rules = uid;
+                        zombie.Settings = rules.EarlySettings;
+                        zombie.VictimSettings = rules.VictimSettings;
+
+                        var mind = player.Data.ContentData()?.Mind;
+                        if (mind?.Session != null)
+                        {
+                            rules.InitialInfectedNames[zombie.BeforeZombifiedEntityName] = mind.Session.Name;
+                        }
+                    }
+                }
+                else
+                {
+	                playerList.Add(player);
+
+	                var pref = (HumanoidCharacterProfile) _prefs.GetPreferences(player.UserId).SelectedCharacter;
+	                    if (pref.AntagPreferences.Contains(rules.PatientZeroPrototypeId))
+	                        prefList.Add(player);
```

Comment:
Please don't mix tabs and spaces.

Reply:
That has to be an artifact of the merge, sorry